core/commands/IssueUpdateCommand.php

```diff
@@ -0,0 +1,489 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+require_api( 'access_api.php' );
+require_api( 'authentication_api.php' );
+require_api( 'bug_api.php' );
+require_api( 'config_api.php' );
+require_api( 'constant_inc.php' );
+require_api( 'custom_field_api.php' );
+require_api( 'date_api.php' );
+require_api( 'email_api.php' );
+require_api( 'error_api.php' );
+require_api( 'event_api.php' );
+require_api( 'file_api.php' );
+require_api( 'helper_api.php' );
+require_api( 'lang_api.php' );
+require_api( 'last_visited_api.php' );
+require_api( 'profile_api.php' );
+require_api( 'relationship_api.php' );
+require_api( 'string_api.php' );
+require_api( 'user_api.php' );
+
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_api.php' );
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_enum_api.php' );
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_issue_api.php' );
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_project_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+/**
+ * Sample:
+ * {
+ *   "query": {
+ *   },
+ *   "payload": {
+ *      "issue": ... see rest issue update documentation
+ *      "bugnote": ... see rest issue update documentation
+ *   },
+ *   "options: {
+ *      "action_type": BUG_UPDATE_TYPE_NORMAL,
+ *      "bug_id": 1234,
+ *   }
+ * }
+ */
+
+/**
+ * A command that updates an issue.
+ */
+class IssueUpdateCommand extends Command {
+
+    /**
+     * The existing issue.
+     *
+     * @var BugData
+     */
+    private $existing_issue = null;
+
+    /**
+     * The updated issue.
+     *
+     * @var BugData
+     */
+    private $updated_issue = null;
+
+    /**
+     * The bugnote.
+     *
+     * @var BugNoteData
+     */
+    private $bug_note = null;
+
+    /**
+     * @var integer
+     */
+    private $user_id;
+
+    /**
+     * The custom fields to set.
+     */
+    private $custom_fields_to_set = array();
+
+    /**
+     * The resolve issue status.
+     *
+     * * @var bool
+     */
+    private $resolve_issue = false;
+
+    /**
+     * The close issue status.
+     *
+     * * @var bool
+     */
+    private $close_issue = false;
+
+    /**
+     * The reopen issue status.
+     *
+     * * @var bool
+     */
+    private $reopen_issue = false;
+
+    /**
+     * Constructor
+     *
+     * @param array $p_data The command data.
+     */
+    function __construct( array $p_data ) {
+        parent::__construct( $p_data );
+    }
+
+    /**
+     * Validate the data.
+     */
+    protected function validate() {
+        $this->user_id = auth_get_current_user_id();
+
+        $f_update_type = $this->option( 'action_type', BUG_UPDATE_TYPE_NORMAL );
```

Comment:
The code inside the function should use `$t_` and not `$f_` prefix, since in this case these variables are now initialized from the payload and not via `gpc_*`.

Reply:
Fixed on offer.